supervisor/host/manager.py

```diff
@@ -95,29 +95,38 @@ def supported_features(self) -> list[HostFeature]:
 
         return features
 
-    async def reload(self):
+    async def reload(
+        self,
+        *,
+        services: bool = True,
+        network: bool = True,
+        agent: bool = True,
+        audio: bool = True,
+    ):
         """Reload host functions."""
         await self.info.update()
 
-        if self.sys_dbus.systemd.is_connected:
+        if services and self.sys_dbus.systemd.is_connected:
             await self.services.update()
 
-        if self.sys_dbus.network.is_connected:
+        if network and self.sys_dbus.network.is_connected:
             await self.network.update()
 
-        if self.sys_dbus.agent.is_connected:
+        if agent and self.sys_dbus.agent.is_connected:
             await self.sys_dbus.agent.update()
 
-        with suppress(PulseAudioError):
-            await self.sound.update()
+        if audio:
+            with suppress(PulseAudioError):
+                await self.sound.update()
 
         _LOGGER.info("Host information reload completed")
         self.supported_features.cache_clear()  # pylint: disable=no-member
 
     async def load(self):
         """Load host information."""
         with suppress(HassioError):
-            await self.reload()
+            await self.reload(network=False)
+            await self.network.load()
```

Comment:
maybe we should call first all platform with `load` and on a new suppress block we can reload it to update data. In that way, we need maybe also no selective reload functionality?

Reply:
Yea I thought about that. I was mostly trying to minimize changes within this PR. None of those other components have a `load` right now so I would have to add one to each. Plus it basically felt like I would be copying everything in the `reload` function and pasting it in another place since I have to do all the same steps.

But I guess that's fine if we think there's value in having a `load` in each since it might do more then just call `update` in the future.

Btw is there a wrapper that checks if `is_connected` returns true before calling the method similar to `dbus_connected` for the `DbusInterface` classes? I couldn't find one. Seems like a better pattern then calling `if is_connected` before every `update`/`load` right?